Products/Zuul/catalog/model_catalog.py

```diff
@@ -0,0 +1,179 @@
+
+import logging
+import zope.component
+
+from interfaces import IModelCatalog
+from zenoss.modelindex import indexed, index
+from zenoss.modelindex.field_types import StringFieldType, \
+     ListOfStringsFieldType, IntFieldType, DictAsStringsFieldType, LongFieldType
+from zenoss.modelindex.constants import INDEX_UNIQUE_FIELD as UID
+from zenoss.modelindex.exceptions import IndexException
+from Products.ZenModel.Software import Software
+from Products.ZenModel.OperatingSystem import OperatingSystem
+from Products.ZenUtils.GlobalConfig import getGlobalConfiguration
+from Products.Zuul.catalog.interfaces import IIndexableWrapper
+from zope.component import getGlobalSiteManager
+from zope.interface import implements
+
+log = logging.getLogger("model_catalog")
+
+#logging.getLogger("requests").setLevel(logging.ERROR) # requests can be pretty chatty 
+
+
+class ModelCatalogIndexable(object):
+
+    implements(IModelCatalog)
+
+    def __init__(self, wrapped):
+        #@TODO HACK I am getting issues with acqusition
+        try:
+            self.wrapped = wrapped.primaryAq()
```

Comment:
This is actually quite expensive. We should trust the wrapped passed in, if at all possible. primaryAq() slows it way down.